x-pack/plugins/observability_solution/inventory/public/components/entities_grid/index.tsx

```diff
@@ -155,6 +160,47 @@ export function EntitiesGrid({
     [onChangeSort]
   );
 
+  const getEntityIcon = useCallback((entity: LatestEntity) => {
+    const entityType = entity[ENTITY_TYPE];
+
+    if (entityType === 'host' || entityType === 'container') {
+      const cloudProvider = entity[CLOUD_PROVIDER] as CloudProvider;
+
+      if (!cloudProvider) return null;
```

Comment:
Same here. CloudProviderIcon handles if the provider is undefined

Reply:
Same as https://github.com/elastic/kibana/pull/194543#discussion_r1782711653.